controllers/ratelimitpolicy_limits.go

```diff
@@ -105,8 +124,137 @@ func (r *RateLimitPolicyReconciler) buildRateLimitIndex(ctx context.Context, rlp
+		// Override is set -> affected policy is Affected by rlp
+		if kuadrantgatewayapi.IsTargetRefGateway(rlp.GetTargetRef()) && rlp.Spec.Overrides != nil {
+			r.AffectedPolicyMap.SetAffectedPolicy(p, []client.ObjectKey{client.ObjectKeyFromObject(rlp)})
+		}
+		_, err := r.reconcileStatus(ctx, p, nil)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *RateLimitPolicyReconciler) getAffectedPoliciesInfo(rlp *kuadrantv1beta2.RateLimitPolicy, t *kuadrantgatewayapi.Topology) ([]kuadrantgatewayapi.Policy, int) {
+	topologyIndexes := kuadrantgatewayapi.NewTopologyIndexes(t)
+	var affectedPolicies []kuadrantgatewayapi.Policy
+	var numUnTargetedRoutes int
+
+	for _, gw := range t.Gateways() {
+		policyList := topologyIndexes.PoliciesFromGateway(gw.Gateway)
+		numUnTargetedRoutes += len(topologyIndexes.GetUntargetedRoutes(gw.Gateway))
+
+		if slices.Contains(utils.Map(policyList, func(p kuadrantgatewayapi.Policy) client.ObjectKey {
+			return client.ObjectKeyFromObject(p)
+		}), client.ObjectKeyFromObject(rlp)) {
+			affectedPolicies = append(affectedPolicies, policyList...)
+		}
+	}
+
+	return affectedPolicies, numUnTargetedRoutes
+}
+
+// applyGatewayOverrides a Gateway RLP is not affected if there is untargetted routes or affects other policies
+// Otherwise, it
+func (r *RateLimitPolicyReconciler) applyGatewayOverrides(logger logr.Logger, rlp *kuadrantv1beta2.RateLimitPolicy, numUnTargetedRoutes int, affectedPolicies []kuadrantgatewayapi.Policy) {
+	if rlp.Spec.Overrides == nil && numUnTargetedRoutes == 0 {
+		r.AffectedPolicyMap.SetAffectedPolicy(rlp, utils.Map(affectedPolicies, func(p kuadrantgatewayapi.Policy) client.ObjectKey {
+			return client.ObjectKeyFromObject(p)
+		}))
+		logger.V(1).Info("policy has no free routes to enforce default policy")
+	} else if rlp.Spec.Overrides != nil && len(affectedPolicies) == 0 && numUnTargetedRoutes == 0 {
+		r.AffectedPolicyMap.SetAffectedPolicy(rlp, []client.ObjectKey{})
+		logger.V(1).Info("policy has no free routes to enforce override policy")
+	} else {
+		r.AffectedPolicyMap.RemoveAffectedPolicy(rlp)
+	}
+}
+
+func (r *RateLimitPolicyReconciler) applyRouteOverrides(logger logr.Logger, rlp *kuadrantv1beta2.RateLimitPolicy, affectedPolicies []kuadrantgatewayapi.Policy) []kuadrantgatewayapi.Policy {
+	filteredPolicies := utils.Filter(affectedPolicies, func(policy kuadrantgatewayapi.Policy) bool {
+		return kuadrantgatewayapi.IsTargetRefGateway(policy.GetTargetRef())
+	})
+
+	sort.Sort(kuadrantgatewayapi.PolicyByTargetRefKindAndCreationTimeStamp(filteredPolicies))
+
+	for _, policy := range filteredPolicies {
+		p := policy.(*kuadrantv1beta2.RateLimitPolicy)
+		if p.Spec.Overrides != nil {
+			rlp.Spec.CommonSpec().Limits = p.Spec.Overrides.Limits
+			logger.V(1).Info("applying overrides from parent policy", "parentPolicy", client.ObjectKeyFromObject(p))
+			r.AffectedPolicyMap.SetAffectedPolicy(rlp, []client.ObjectKey{client.ObjectKeyFromObject(p)})
+			break
```

Comment:
No, I went with `break` because the applied overrides currently can only come from one gateway RLP, with the oldest having the precdence. In this case, we should break out the loop

Reply:
Of course it is sorted by the time stamp. Makes perfect sense now.